src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -50,12 +71,20 @@ private void updateBootstrapServers() {
             return;
         }
 
-        if (!brokers.isEmpty()) {
-            kafkaProperties.setProperty(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
-                    brokers.stream().map(Broker::toString).collect(Collectors.joining(",")));
-            LOG.info("Kafka client bootstrap servers: {}",
-                    kafkaProperties.get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG));
+        if (brokers.isEmpty()) {
+            return;
+        }
+        final String bootstrapServers = brokers.stream().map(Broker::toString).collect(Collectors.joining(","));
+        final String currentBootstrapServers =
+                (String) kafkaProperties.get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG);
+
+        if (bootstrapServers.equals(currentBootstrapServers)) {
+            return;
         }
+
+        kafkaProperties.setProperty(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        this.ipAddressChangeListeners.forEach(Runnable::run);
```

Comment:
In case if someone will throw exception from here - singlethreadedscheduled executor will die and will never ever execute anything.

Reply:
I feel that singlethreadedscheduled is a simple german word